libethereum/ValidationSchemes.cpp

```diff
@@ -0,0 +1,142 @@
+string const c_daoHardforkBlock = "daoHardforkBlock";
+string const c_EIP150ForkBlock = "EIP150ForkBlock";
+string const c_EIP158ForkBlock = "EIP158ForkBlock";
+string const c_byzantiumForkBlock = "byzantiumForkBlock";
+string const c_eWASMForkBlock = "eWASMForkBlock";
+string const c_constantinopleForkBlock = "constantinopleForkBlock";
+string const c_accountStartNonce = "accountStartNonce";
+string const c_maximumExtraDataSize = "maximumExtraDataSize";
+string const c_tieBreakingGas = "tieBreakingGas";
+string const c_blockReward = "blockReward";
+string const c_difficultyBoundDivisor = "difficultyBoundDivisor";
+string const c_minimumDifficulty = "minimumDifficulty";
+string const c_durationLimit = "durationLimit";
+string const c_chainID = "chainID";
+string const c_networkID = "networkID";
+string const c_allowFutureBlocks = "allowFutureBlocks";
+
+void validateConfigJson(js::mObject const& _obj)
+{
+    requireJsonFields(_obj, "ChainParams::loadConfig",
+        {{c_sealEngine, {{json_spirit::str_type}, JsonFieldPresence::Required}},
+            {c_params, {{json_spirit::obj_type}, JsonFieldPresence::Required}},
+            {c_genesis, {{json_spirit::obj_type}, JsonFieldPresence::Required}},
+            {c_accounts, {{json_spirit::obj_type}, JsonFieldPresence::Required}}});
+
+    requireJsonFields(_obj.at(c_genesis).get_obj(), "ChainParams::loadConfig::genesis",
+        {{c_author, {{json_spirit::str_type}, JsonFieldPresence::Required}},
+            {c_nonce, {{json_spirit::str_type}, JsonFieldPresence::Required}},
+            {c_gasLimit, {{json_spirit::str_type}, JsonFieldPresence::Required}},
+            {c_timestamp, {{json_spirit::str_type}, JsonFieldPresence::Required}},
+            {c_difficulty, {{json_spirit::str_type}, JsonFieldPresence::Required}},
+            {c_extraData, {{json_spirit::str_type}, JsonFieldPresence::Required}},
+            {c_mixHash, {{json_spirit::str_type}, JsonFieldPresence::Required}},
+            {c_parentHash, {{json_spirit::str_type}, JsonFieldPresence::Optional}}});
+
+    js::mObject const& accounts = _obj.at(c_accounts).get_obj();
+    for (auto const& acc : accounts)
+        validateAccountObj(acc.second.get_obj());
+}
+
+void validateAccountMapObj(js::mObject const& _obj)
+{
+    // A map object with possibly defined fields
+    requireJsonFields(_obj, "validateAccountMapObj",
+        {{c_storage, {{json_spirit::obj_type}, JsonFieldPresence::Optional}},
+            {c_balance, {{json_spirit::str_type}, JsonFieldPresence::Optional}},
+            {c_nonce, {{json_spirit::str_type}, JsonFieldPresence::Optional}},
+            {c_code, {{json_spirit::str_type}, JsonFieldPresence::Optional}},
+            {c_precompiled, {{json_spirit::obj_type}, JsonFieldPresence::Optional}},
+            {c_shouldnotexist, {{json_spirit::str_type}, JsonFieldPresence::Optional}},
+            {c_wei, {{json_spirit::str_type}, JsonFieldPresence::Optional}}});
+    }
```

Comment:
Indentation is off here. Use clang-format

Reply:
I used clang-format